src/transformers/integrations/integration_utils.py

```diff
@@ -1071,6 +1071,8 @@ def on_log(self, args, state, control, logs, model=None, **kwargs):
             for k, v in logs.items():
                 if isinstance(v, (int, float)):
                     metrics[k] = v
+                elif isinstance(v, torch.Tensor) and v.numel() == 1:
+                   metrics[k] = float(v)
```

Comment:
Let's use `.item()` here so we preserve if it's a float or int 

```suggestion
                   metrics[k] = v.item()
```